### PR TITLE
Fixes several things about Survey List

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
Addresses #6333 

Aligns Select All with other select buttons
Add required asterisk to those items that are required
Adds label for the Default and Question Type column
Adds chips for multiselect items.
Adds RBAC to add and edit survey.
Also fixes a bug where the survey was not reloading properly after edit

Be aware of this bug https://github.com/ansible/awx/issues/6561
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.3
```

##### ADDITIONAL INFORMATION
Screen Shot 2020-04-03 at 10 38 25 AM
 
I'd especially like feedback on how I am handling errors, specifically in ErrorDetail.jsx

### CODE REVIEW

File: awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx [hunk 1]
```diff
@@ -73,19 +86,51 @@ function SurveyListItem({
...
+                    aria-hidden="true"
+                  >
+                    *
+                  </Required>
+                )}
+              </>
+            </DataListCell>,
+
+            <DataListCell key="type">
+              <Label>{i18n._(t`Type`)}</Label>
```
Comment (dsesami) [inline awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx]: Can this be `Type:` with a colon? It looks better like that.

File: awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx [hunk 2]
```diff
@@ -73,19 +86,51 @@ function SurveyListItem({
...
+              {question.type}
+            </DataListCell>,
+            <DataListCell key="default">
+              <Label>{i18n._(t`Default`)}</Label>
```
Comment (dsesami) [inline awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx]: Adding a colon to this as well would help its appearance. `Default:`

**Review by dsesami — CHANGES_REQUESTED** [review]
- Format of multiple choice answers is not clear. The user has no way of telling if they are supposed to write commas or newlines as delimiters. 
- There is also no way to set a multiselect Default with more than one answer; this should be a TextArea if you need newlines.. If the user attempts to use commas as delimiters, they are met with a 400 error and no further information.

File: awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx (hunk 2 above)
Comment (AlexSCorey) [reply awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx:118]: > * Format of multiple choice answers is not clear. The user has no way of telling if they are supposed to write commas or newlines as delimiters.

Will add a tooltip to this field to let the user know each answer must be on its own line.  

> * There is also no way to set a multiselect Default with more than one answer; this should be a TextArea if you need newlines.. If the user attempts to use commas as delimiters, they are met with a 400 error and no further information.

@dsesami if the user uses new lines they should for each answer choice and each default answer they should see a chip for each answer choice
Screen Shot 2020-04-06 at 10 35 02 AM

**Comment by AlexSCorey** [comment]
> * Format of multiple choice answers is not clear. The user has no way of telling if they are supposed to write commas or newlines as delimiters.

Will add a tooltip to address this.  
> * There is also no way to set a multiselect Default with more than one answer; this should be a TextArea if you need newlines.. If the user attempts to use commas as delimiters, they are met with a 400 error and no further information.

@dsesami if the user uses new lines they should for each answer choice and each default answer they should see a chip for each answer choice
Screen Shot 2020-04-06 at 10 35 02 AM

**Review by dsesami — CHANGES_REQUESTED** [review]
Comments pasted here:

> Survey limits don't seem to be honored.
> 
> 1. Create a text survey named foo
> 2. Make the default value bar
> 3. Set the character limit max to 1
> 4. 
> 
> I would expect this to not be allowed.
> 
> 
> 
> Trying to turn on the survey from the list page when the JT is missing a project or inventory results in a 400 with no further information.
> 
>

**Comment by dsesami** [comment]
One more comment; if a default value in a multiple choice problem is not one of the available options, it 400s without proper explanation. We do get the answer from the API, so we should have that shown in the UI.

```{"error":"Default choice must be answered from the choices listed."}```

**Review by dsesami — APPROVED** [review]
Excellent!

File: awx/ui_next/src/components/ErrorDetail/ErrorDetail.jsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ class ErrorDetail extends Component {
       message =
         typeof response.data === 'string'
           ? response.data
-          : response.data?.detail;
+          : response.data?.detail || Object.values(error.response.data)[0];
```
Comment (keithjgrant) [inline awx/ui_next/src/components/ErrorDetail/ErrorDetail.jsx]: A couple things here:

* If there's no `data.detail`, but data is an object of keys/error messages, we should probably show all the error messages, not just the first (`[0]`). Maybe show them in a `<ul>`?
* if `response.data` is undefined, this will throw an error when it tries to `Object.values(error.response.data)`

I think you'll need to break apart this ternary into a more complete set of possible ways to parse the response.data into a string. It would probably worth TDD-driving a helper function to do it and ensure we handle all sorts of cases appropriately.

File: awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx [hunk 4]
```diff
@@ -27,8 +28,20 @@ const Button = styled(_Button)`
...
+const Chip = styled(_Chip)`
+  margin-right: 5px;
+`;
+
+const Label = styled.b`
+  margin-right: 20px;
+`;
 
 function SurveyListItem({
+  canAddAndEditSurvey,
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx]: just a nitpick, but you could probably shorten this to just `canEdit` since there's no "adding" going on in the individual items (and we already know it's a survey)

File: awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx [hunk 5]
```diff
@@ -27,8 +28,20 @@ const Button = styled(_Button)`
   padding-bottom: 0;
   padding-left: 0;
 `;
+const Required = styled.span`
+  color: red;
+  margin-left: 5px;
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx]: use the PF variables to stick with their theming

```
color: var(--pf-global--danger-color--100);
margin-left: var(--pf-global--spacer--xs);
```

File: awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx [hunk 6]
```diff
@@ -27,8 +28,20 @@ const Button = styled(_Button)`
...
+`;
+const Chip = styled(_Chip)`
+  margin-right: 5px;
+`;
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Template/Survey/SurveyListItem.jsx]: Can you use a ChipGroup to contain these instead of adding custom styles? It should take care the spacing for you

**Review by keithjgrant — COMMENTED** [review]
Nice job tackling these things. I'm still looking through this, but here are a few thoughts you can look into while I do

**Review by keithjgrant — APPROVED** [review]
Nice work!

**Comment by AlexSCorey** [comment]
recheck

**Comment by AlexSCorey** [comment]
recheck

**Comment by AlexSCorey** [comment]
regate

**Comment by AlexSCorey** [comment]
recheck

**Comment by AlexSCorey** [comment]
recheck

**Comment by AlexSCorey** [comment]
regate

**Comment by AlexSCorey** [comment]
recheck
